js/test/main.test.js

```diff
@@ -344,6 +344,183 @@ describe("sqlparser", function () {
+      const sql = await globalThis.sqlparser.createStatementFromObject(obj);
+      const { statements } = await globalThis.sqlparser.normalize(sql);
+      strictEqual(sql, statements[0]);
+    })
+
+    test("from object", async function() {
+      /** @type {import("@tableland/sqlparser").CreateTable } */
+      const obj = {
+        Table: {
+          Name: "blah_5_30001",
+          IsTarget: true
+        },
+        ColumnsDef: [
+          {
+            Column: {
+              Name: "id",
+              TableRef: null,
+            },
+            Type: "int",
+            Constraints: []
+          }
+        ],
+        Constraints: [],
+        StrictMode: false
+      };
+      const observed = await globalThis.sqlparser.createStatementFromObject(obj);
+      const expected = "create table blah_5_30001(id int)";
+      strictEqual(observed, expected);
+    });
+
+    test("from object with a constraint", async function() {
+      /** @type {import("@tableland/sqlparser").CreateTable } */
+      const obj = {
+        Table: {
+          Name: "blah_5_30001",
+          IsTarget: true
+        },
+        ColumnsDef: [
+          {
+            Column: {
+              Name: "id",
+              TableRef: null,
+            },
+            Type: "int",
+            Constraints: [{
+              Type: "primary-key",
+              Name: "id",
+              Order: "asc",
+              AutoIncrement: false,
+            }]
+          }
+        ],
+        Constraints: [],
+        StrictMode: false
+      };
+      const observed = await globalThis.sqlparser.createStatementFromObject(obj);
+      const expected = "create table blah_5_30001(id int constraint id primary key asc)";
+      strictEqual(observed, expected);
+    });
+
+    test("from object with ambiguous constraints", async function() {
+      /** @type {import("@tableland/sqlparser").CreateTable } */
+      const obj = {
+        Table: {
+          Name: "blah_5_",
+          IsTarget: true
+        },
+        ColumnsDef: [
+          {
+            Column: {
+              Name: "id",
+              TableRef: null,
+            },
+            Type: "int",
+            Constraints: [{
+              Type: "not-null",
+              "Name": "",
+            },
+            {
+              Type: "unique",
+              "Name": "",
+            }]
+          }
+        ],
+        Constraints: [],
+        StrictMode: false
+      };
+      const observed = await globalThis.sqlparser.createStatementFromObject(obj);
+      const expected = "create table blah_5_(id int not null unique)";
+      strictEqual(observed, expected);
+    });
+
+    test("from schema object to create table to ast object", async function() {
+      /**
+       * 
+       * @param {string} tableName 
+       * @param {any} schema 
+       * @returns {string} The CREATE statement string
+       */
+      function generateCreateTableStatement(tableName, schema) {
```

Comment:
This function converts from your Schema format into a create statement string @asutula. From there, we can generate the more verbose/complete AST format. However, we _can't_ round-trip any constraints that include an expression, only things like primary key, unique, etc. Not check, generated as, and others.